### PR TITLE
PBI 20 & 21: Sidebar and Layout

### LINKED ISSUE #47: Sidebar

### LINKED ISSUE #48: PBI-21: Course layout

### PR DESCRIPTION
**Pull request by Zsaschz**
Done

- Sidebar
- Course layout
- Global snackbar
- Refactor all courses related pages to use layout (course stream, people, viewcourse, create course)

### Sidebar
image
image

### Course Layout
image

### To use layout
image
This will place your components inside the slot in courseStream.vue

### Snackbar Usage
if you use AppLayout.vue, you can open snackbar by calling
```this.$store.dispatch("OPEN_SNACKBAR", mesesage)```

### CODE REVIEW

File: src/resources/app/pages/courseStream.vue [hunk 1]
```diff
@@ -1,15 +1,79 @@
...
 <script>
+import AppLayout from "../components/Layout/AppLayout.vue";
 export default {
+  layout: [AppLayout],
+  data() {
+    return {
+      tabs: [
+        {
+          name: "Classwork",
+          href: "/classwork",
```
Comment (fxdros) [inline src/resources/app/pages/courseStream.vue:67]: Are we going to redirect the user after clicking any tab option or just hide/display the content?

File: src/resources/app/pages/courseStream.vue [hunk 2]
```diff
@@ -1,15 +1,79 @@
...
+        },
+        {
+          name: "People",
+          href: "/people",
+        },
+        {
+          name: "Main Page",
```
Comment (fxdros) [inline src/resources/app/pages/courseStream.vue]: If you don't mind, please change `Main Page` to `Announcements` (don't forget to use plural form), and please change the order to :
1. Announcements
2. Classwork
3. People

Comment (Zsaschz) [reply src/resources/app/pages/courseStream.vue]: okay, i was referencing from figma earlier

File: src/resources/app/pages/courseStream.vue (hunk 1 above)
Comment (Zsaschz) [reply src/resources/app/pages/courseStream.vue:67]: currently, our implementation requires us to redirect, but with inertia there won't be any loading animation and works just like an SPA. though it is still a bit slow and we don't actually know why. another implementation is to use vue-router but i think it is better to test it on a seperate branch before actually implementing it.

Comment (fxdros) [reply src/resources/app/pages/courseStream.vue:67]: Alright then, I guess it's okay for now to do it this way

**Review by fxdros — APPROVED** [review]
Everything looks good to me, keep it up!

**Review by IlmaMannix — APPROVED** [review]
Τhanks for your hard work

**Pull request merged**